lisa-kernel/src/main/scala/lisa/kernel/proof/SCProofChecker.scala

```diff
@@ -225,16 +233,20 @@ object SCProofChecker {
               else SCInvalidProof(SCProof(step), Nil, "Right-hand side of conclusion + ψ must be same as right-hand side of premise + φ⇒ψ.")
             else SCInvalidProof(SCProof(step), Nil, "Left-hand side of conclusion + psi must be same as left-hand side of premise.")
           /*
-           *  Γ |- a⇒ψ, Δ    Σ |- ψ⇒φ, Π
+           *  Γ |- φ⇒ψ, Δ    Σ |- ψ⇒φ, Π
            * ----------------------------
-           *      Γ, Σ |- φ⇔b, Π, Δ
+           *      Γ, Σ |- φ⇔ψ, Π, Δ
            */
           case RightIff(b, t1, t2, phi, psi) =>
             val phiImpPsi = ConnectorFormula(Implies, Seq(phi, psi))
             val psiImpPhi = ConnectorFormula(Implies, Seq(psi, phi))
             val phiIffPsi = ConnectorFormula(Iff, Seq(phi, psi))
             if (isSameSet(b.left, ref(t1).left union ref(t2).left))
-              if (isSameSet(b.right + phiImpPsi + psiImpPhi, ref(t1).right union ref(t2).right + phiIffPsi))
+              if (
+                isSubset(ref(t1).right, b.right + phiImpPsi) &&
+                isSubset(ref(t2).right, b.right + psiImpPhi) &&
+                isSubset(b.right, ref(t1).right union ref(t2).right + phiIffPsi)
```

Comment:
I was adapting this to basic step tactics, and noticed that this is perhaps too restrictive? We should have a more uniform policy on where and how much we allow weakening. 

It is perhaps enough to check that $\phi \iff \psi \in bot.right$.

Reply:
Maybe, but it's the same semantic as what we had. In general we don't permit arbitrary weakening in non-Weakening proof steps.